### DISCUSSION TITLE
Linear interpolation formula seems funky for fvm - am I interpreting correctly?

### DISCUSSION THREAD

**Discussion by jelc53** [pybamm-team/PyBaMM#2944] [Q&A] [answered]
I would have thought we would try to estimate surface concentration by 'adding' our linear gradient step to the fN (concentration value at final cell interior node). Am I interpreting this correctly for 'right' boundary and 'linear' extrapolation ? 

Code here (formula in comment of line 914): https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/spatial_methods/finite_volume.py#L813 

![image](https://github.com/pybamm-team/PyBaMM/assets/30467615/73247dc8-bc4f-4fa6-8fc4-f736e873a89d)

**Comment by jelc53** [comment]
FWIW, I think it should be f(x*) = f_N + (dxN / dxNm1) (f_N - f_Nm1) instead of # f(x*) = f_N - (dxN / dxNm1) (f_N - f_Nm1). I think that is how it is implemented in the CSR matrix code.

**Reply by valentinsulzer** [reply, marked as answer]
This 
```
csr_matrix(
                            (
                                [-(dxN / dxNm1), 1 + (dxN / dxNm1)],
                                ([0, 0], [prim_pts - 2, prim_pts - 1]),
                            ),
                            shape=(1, prim_pts),
                        )
```
is `f_N + (dxN/dxNm1) * (f_N-f_Nm1)`

**Reply by jelc53** [reply]
Oh so there is a typo in the comment then! It says `f(x*) = f_N - (dxN / dxNm1) (f_N - f_Nm1)` when it should be `f(x*) = f_N + (dxN/dxNm1) * (f_N-f_Nm1)` as you say (minus flips to plus). This makes sense to me since it would be logically equivalent to the neumann boundary (use_bcs) version above in the code (`f(x*) = f_N + dxN * f'(x*)`).
